libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1297,6 +1297,11 @@ bool AP_GPS::get_first_external_instance(uint8_t& instance) const
 void AP_GPS::handle_external(const AP_ExternalAHRS::gps_data_message_t &pkt, const uint8_t instance)
 {
     if (get_type(instance) == GPS_TYPE_EXTERNAL_AHRS && drivers[instance] != nullptr) {
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        if (pkt.longitude == 0 && pkt.latitude == 0) {
```

Comment:
Any reason you're not checking altitude here?  It's traditional.... and non-zero alt is suffucient to make a location valid as far as ArduPilot's concerned...

Reply:
Because the bug I found had altitude set, but lat/lon were not set due to a division/multiplication error of 1E7.